docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1135 @@
+
+|     |     |
+|-----|-----|
+|**Pros** | Execution time is fast.|  
+|**Cons** | Profile data is not updated in real time if user edits it in text editor while running The Schwarzenegger.|
+
+[&#8593; Return to Top](#developer-guide)
+
+#### 4.2.3. <a id="editing-a-profile">Editing a Profile</a>
+This feature allows user to anytime go back to edit a profile created in the past such as editing physique data and expected daily calories intake. The failure to do so will trigger an exception where the user will be notified of the reason, e.g. invalid command format. The action will be aborted, and the program will advise the user to type "help" for command syntax reference. 
+
+If the editing is successful, a confirmation message on the edited profile will be displayed to the user. 
+
+**Implementation**
+
+When the user attempts to edit a profile, the ProfileSession, Ui, ProfileParser, Command, CommandLib, ProfileStorage, Profile and CommandResult classes will be accessed, and the following sequence of actions is called to prompt execution result to user:
+
+1. User executes `edit /w 60`
+    1. `ProfileSession` calls `Ui.getCommand()` to receive user input.
+    1. `ProfileSession` calls `ProfileParser.parseCommand()` to parse user input into a string array.
+1. Creating `ProfileEdit` object.
+    1. Based on the parsed input, `ProfileSession` calls `CommandLib` to return the correct Command Object `ProfileEdit`.
+1. Executing command.
+    1. `ProfileSession` calls `ProfileEdit.execute()` with the rest of parsed input.
+    1. `ProfileEdit` calls `ProfileStorage.loadData()` to load existing profile in the system. If there is no existing profile, `ProfileAdd` returns a failure result to `ProfileSession`. Otherwise, the process continues with step `iii`.
+    1. `ProfileEdit` calls `ProfileParser.extractCommandTagAndInfo()` to parse user input into specific tags and information.
+    1. Based on the parsed information from `ProfileParser.extractCommandTagAndInfo()`, `ProfileEdit` creates a new `Profile`.
+    1. `ProfileEdit` calls `Profile.equals()` to compare the edited and existing profile. If there are no changes, `ProfileEdit` returns a failure result to `ProfileSession`. Otherwise, the process continues with step `vi`.
+    1. `ProfileEdit` calls `ProfileStorage.saveData()` to save the edited `Profile` object.
+    1. `ProfileAdd` returns a successful result to `ProfileSession`.
+1. Prompting result to user.
+    1. `ProfileSession` calls `CommandResult.getFeedbackMessage()` to get the execution feedback message.
+    1. `ProfileSession` calls `Ui.showToUser()` to show result to the user.
+
+All descriptions, warnings and responses will be handled by `Ui` to ensure consistence across the app.
+
+The sequence diagram below summarizes how creating a new profile works:
+
+![Load Data Sequence Diagram](pictures/khoa/EditProfile.png)
```

Comment:
The sequence diagram seems to be bordering on too complicated. Would it be better to omit some less important(repetitive) parts?
<img width="841" alt="36201BAC-6479-471C-8C97-230A2B58C2F0" src="https://user-images.githubusercontent.com/57485187/97533797-d79bee80-19f3-11eb-898b-fcaf34f4db95.png">